File: src/test/console/dotnet/MobileAutomation/Tests/Scripts/IOSAppTest.cs
using System;
using System.Threading;
using MobileAutomation.Config;
using MobileAutomation.Service;
using MobileAutomation.Service.Model;
using MobileAutomation.Tests.Utils;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Remote;
using Xunit;

namespace MobileAutomation.Tests.Scripts
{
  public class IOSAppTest : BaseTest
  {
    public IOSAppTest()
    {
      var device = new DeviceService().GetOnlineDeviceSync(DeviceType.IOS);
      Console.WriteLine(string.Format("Device {0} osVersion: {1} udid: {2}",
                                         device.DeviceName,
                                         device.PlatformVersion,
                                         device.Udid));
      driver = new RemoteWebDriver(AutomationUtils.Instance.KobitonServerUrl,
                                     AutomationUtils.Instance.DesiredCapabilitiesIOSApp(device),
                                     TimeSpan.FromSeconds(Configs.SESSION_TIMEOUT));
      driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(60);
    }

    [Fact]
    public void TestIFixitIOSApp()
    {
      try
      {
        TestNavigationOnAcuraSupportCommunity();
      }
      finally
      {
        this.Dispose();
      }
    }

    public void TestNavigationOnAcuraSupportCommunity()
    {
      Console.WriteLine("should allow to navigate to some devices on Acura Support Community");
      /*
      * Steps:
      * 1. Click on "Car and Truck" Categories on Homepage
      * 2. Click on "Acura" Categories
      *
      * Expected:
      * 1. General Information is "Acura".
      * 2.Verify Acura Integra displays.
      */
      Thread.Sleep(3000);
      driver.FindElementByXPath("//XCUIElementTypeButton[@name='START A REPAIR']").Click();
      Thread.Sleep(2000);
      driver.FindElementByXPath("//*[@name='Car and Truck']").Click();
      Thread.Sleep(2000);
      driver.FindElementByXPath("//*[@name='Acura']").Click();
      Thread.Sleep(2000);
      WebDriverWait wait = new WebDriverWait(driver, new TimeSpan(0, 0, 60));
      wait.Until(ExpectedConditions.ElementToBeClickable(By.XPath("//XCUIElementTypeNavigationBar")));
      string acuraText = driver
        .FindElementByXPath("//XCUIElementTypeNavigationBar").GetAttribute("name");
      bool hasAcuraIntegra = driver
        .FindElementByXPath("//XCUIElementTypeStaticText[@name='Acura Integra']").Displayed;
      Assert.Equal(acuraText, "Acura");
      Assert.Equal(hasAcuraIntegra, true);
    }
  }
}

File: src/test/console/java/src/test/java/kobiton/com/testng/iOSAppTest.java
package kobiton.com.testng;

import com.kobiton.service.DeviceService;
import com.kobiton.service.model.Device;
import java.util.List;
import java.util.concurrent.TimeUnit;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;
import io.appium.java_client.ios.IOSDriver;
import java.io.IOException;
import kobiton.com.utils.AutomationUtils;

public class iOSAppTest {

    public static IOSDriver<WebElement> driver = null;

    @BeforeTest
    public void Setup() {
        try {
            Device device = new DeviceService().getOnlineDevice("iOS");
            System.out.println(String.format("Execute with: %s  udid: %s", device.getDeviceName(), device.getUdid()));
            driver = new IOSDriver<>(AutomationUtils.kobitonServerUrl(), AutomationUtils.desiredCapabilitiesiOSApp(device));
            driver.manage().timeouts().implicitlyWait(60, TimeUnit.SECONDS);
            String kobitonSessionId = driver.getSessionDetails().get("kobitonSessionId").toString();
            System.out.println("https://portal.kobiton.com/sessions/" + kobitonSessionId);
        } catch (IOException ex) {
            Assert.fail("set up failed", ex);
        }
    }

    @AfterTest
    public void Teardown() {
        try {
            if (driver != null) {
                driver.quit();
            }
        } catch (Exception e) {
            //Do-nothing
        }
    }

    @Test
    public void test() {
        driver.findElementByXPath("//XCUIElementTypeButton[@name='START A REPAIR']").click();
    }

    public void sleep(int seconds) {
        try {
            Thread.sleep(seconds * 1000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}
